src/utils.ts

```diff
@@ -333,3 +337,46 @@ export function isCursorChange(change: TextDocumentContentChangeEvent, cursor: P
     }
     return false;
 }
+
+type LegacySettingName = "neovimPath" | "neovimInitPath";
+type SettingPrefix = "neovimExecutablePaths" | "neovimInitVimPaths"; //this needs to be aligned with setting names in package.json
+type Platform = "win32" | "darwin" | "linux";
+
+function getSystemSpecificSetting(
+    settingPrefix: SettingPrefix,
+    legacySetting: { environmentVariableName?: "NEOVIM_PATH"; vscodeSettingName: LegacySettingName },
+): string | undefined {
+    const settings = workspace.getConfiguration(EXT_NAME);
+    const isUseWindowsSubsystemForLinux = settings.get("useWSL");
+
+    //https://github.com/microsoft/vscode/blob/master/src/vs/base/common/platform.ts#L63
+    const platform = process.platform as "win32" | "darwin" | "linux";
+
+    const legacyEnvironmentVariable =
+        legacySetting.environmentVariableName && process.env[legacySetting.environmentVariableName];
+
+    //some system specific settings can be loaded from process.env and value from env will override setting value
+    const legacySettingValue = legacyEnvironmentVariable || settings.get(legacySetting.vscodeSettingName);
+    if (legacySettingValue) {
+        return legacySettingValue;
+    } else if (isUseWindowsSubsystemForLinux && platform !== "darwin") {
```

Comment:
I think there should be platform === "win32" only since you can't use WSL on native linux 😄

Reply:
Fixed. I'm not very smart you know lol